services/innkeeper/api/resources/tenant.py

```diff
@@ -0,0 +1,147 @@
+import uuid
+
+import requests
+from flask import request
+from flask_restful import Resource
+from http import HTTPStatus
+
+from marshmallow import ValidationError
+from webargs import fields
+from webargs.flaskparser import use_kwargs
+
+from api.models import Tenant, AccessKey
+from api.schemas.access_key import AccessKeyCreatedSchema, AccessKeyPaginationSchema
+from api.schemas.tenant import TenantSchema, TenantPaginationSchema, TenantCreatedSchema
+from api.utils import hash_password
+
+from flask import current_app
+
+resource_schema = TenantSchema()
+resource_new_schema = TenantCreatedSchema()
+resource_list_schema = TenantSchema(many=True)
+resource_pagination_schema = TenantPaginationSchema()
+
+access_key_new_schema = AccessKeyCreatedSchema()
+access_key_pagination_schema = AccessKeyPaginationSchema()
+
+
+class TenantListResource(Resource):
+    @use_kwargs(
+        {
+            "q": fields.Str(missing=""),
+            "page": fields.Int(missing=1),
+            "per_page": fields.Int(missing=20),
+            "sort": fields.Str(missing="created_at"),
+            "order": fields.Str(missing="desc"),
+        }
+    )
+    def get(self, q, page, per_page, sort, order):
+
+        if sort not in ["created_at", "name"]:
+            sort = "created_at"
+
+        if order not in ["asc", "desc"]:
+            order = "desc"
+
+        paginated_items = Tenant.get_all_active(q, page, per_page, sort, order)
+
+        return resource_pagination_schema.dump(paginated_items), HTTPStatus.OK
+
+    def post(self):
+
+        json_data = request.get_json()
+
+        data = resource_schema.load(data=json_data)
+        try:
+            data = resource_schema.load(data=json_data)
+        except ValidationError as err:
+            return {
+                "message": "Validation errors",
+                "errors": err.messages,
+            }, HTTPStatus.BAD_REQUEST
+
+        tenant = Tenant(**data)
+        if not Tenant.get_by_name(tenant.name):
+
+            # create a tenant in acapy - move to service
+            url = f"{current_app.config['ACAPY_ADMIN_URL']}/multitenancy/wallet"
+            headers = {
```

Comment:
We'll want to refactor this at some point, but I think this is fine for the initial commit.

Reply:
yes, lots of refactoring and cleanup to do - LOTS.